server/routes/visitJourney/dateAndTime.test.ts

```diff
@@ -488,6 +499,34 @@ testJourneys.forEach(journey => {
             expect(auditService.reservedVisit).not.toHaveBeenCalled()
           })
       })
+
+      it('should re-direct to overbooking page if session has no available slots', () => {
+        return request(sessionApp)
+          .post(`${journey.urlPrefix}/select-date-and-time`)
+          .send('visit-date-and-time=4')
+          .expect(302)
+          .expect('location', `${journey.urlPrefix}/select-date-and-time/overbooking`)
+          .expect(() => {
+            expect(visitSessionData.visitSlot).toEqual(<VisitSlot>{
+              id: '4',
+              sessionTemplateReference: 'a1b.2cd.3e4',
+              prisonId,
+              startTimestamp: '2022-02-14T15:30:00',
+              endTimestamp: '2022-02-14T16:35:00',
+              availableTables: 0,
+              capacity: 5,
+              visitRoom: 'room name',
+              visitRestriction: 'OPEN',
+            })
+            expect(visitSessionData.applicationReference).not.toBeDefined()
+
+            expect(
+              journey.isUpdate ? visitService.createVisitApplicationFromVisit : visitService.createVisitApplication,
+            ).toHaveBeenCalledTimes(0)
+            expect(visitService.changeVisitApplication).toHaveBeenCalledTimes(0)
```

Comment:
You could also make the same assertion using `expect(...).not.toHaveBeenCalled()` which is perhaps marginally clearer. Just flagging for info; a matter of personal preference really.